cli/command/load_balancer_test.go

```diff
@@ -0,0 +1,491 @@
+package command
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/quintilesims/layer0/common/models"
+	"github.com/urfave/cli"
+)
+
+func TestAddPortToLoadBalancer(t *testing.T) {
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+
+	base.Resolver.EXPECT().
+		Resolve("load_balancer", "lname").
+		Return([]string{"lid"}, nil)
+
+	ports := []models.Port{
+		models.Port{
+			ContainerPort: 80,
+			HostPort:      80,
+			Protocol:      "tcp",
+		},
+	}
+	healthCheck := models.HealthCheck{}
+
+	loadBalancer := &models.LoadBalancer{
+		HealthCheck:      healthCheck,
+		LoadBalancerID:   "lid",
+		LoadBalancerName: "lname",
+		Ports:            ports,
+	}
+
+	base.Client.EXPECT().
+		ReadLoadBalancer("lid").
+		Return(loadBalancer, nil)
+
+	ports = append(ports, models.Port{
+		ContainerPort: 81,
+		HostPort:      81,
+		Protocol:      "tcp",
+	})
+
+	req := models.UpdateLoadBalancerRequest{
+		HealthCheck:    nil,
+		LoadBalancerID: "lid",
+		Ports:          &ports,
+	}
+
+	base.Client.EXPECT().
+		UpdateLoadBalancer(req).
+		Return("jid", nil)
+
+	job := &models.Job{
+		Status: "Completed",
+		Result: "lid",
+	}
+
+	base.Client.EXPECT().
+		ReadJob("jid").
+		Return(job, nil)
+
+	base.Client.EXPECT().
+		ReadLoadBalancer("lid").
+		Return(&models.LoadBalancer{}, nil)
+
+	loadBalancerCommand := NewLoadBalancerCommand(base.Command())
+	c := getCLIContext(t, args{"lname", "81:81/tcp"}, flags{})
+	if err := loadBalancerCommand.addport(c); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestAddPortToLoadBalancer_UserInputError(t *testing.T) {
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+	loadBalancerCommand := NewLoadBalancerCommand(base.Command())
+
+	contexts := map[string]*cli.Context{
+		"Missing NAME arg": getCLIContext(t, nil, nil),
+		"Missing PORT arg": getCLIContext(t, args{"name"}, nil),
+	}
+
+	for name, c := range contexts {
+		if err := loadBalancerCommand.addport(c); err == nil {
+			t.Fatalf("%s: error was nil!", name)
+		}
+	}
+}
+
+func TestCreateLoadBalancer(t *testing.T) {
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+
+	base.Resolver.EXPECT().
+		Resolve("environment", "ename").
+		Return([]string{"eid"}, nil)
+
+	ports := []models.Port{
+		models.Port{
+			ContainerPort: 80,
+			HostPort:      80,
+			Protocol:      "tcp",
+		},
+	}
+	healthCheck := models.HealthCheck{
+		Target: "TCP:80",
+	}
+	req := models.CreateLoadBalancerRequest{
+		LoadBalancerName: "lname",
+		EnvironmentID:    "eid",
+		IsPublic:         true,
+		Ports:            ports,
+		HealthCheck:      healthCheck,
+	}
+
+	base.Client.EXPECT().
+		CreateLoadBalancer(req).
+		Return("jid", nil)
+
+	job := &models.Job{
+		Status: "Completed",
+		Result: "lid",
+	}
+
+	base.Client.EXPECT().
+		ReadJob("jid").
+		Return(job, nil)
+
+	loadBalancer := &models.LoadBalancer{
+		HealthCheck: healthCheck,
+		Ports:       ports,
+	}
+
+	base.Client.EXPECT().
+		ReadLoadBalancer("lid").
+		Return(loadBalancer, nil)
```

Comment:
[nit] this could just be `Return(&models.LoadBalancer{}, nil)`

...perhaps I'm thinking too much into this, but as someone reading this test it would be very clear to me that `Return(&models.LoadBalancer{}, nil)` means "it doesn't really matter what gets returned in this call" vs. right now it looks like having the `HealthCheck` and `Ports` field set is important.

Reply:
:check: